### ISSUE TITLE
[Object Detection (but not only)][Feature Request] Batch: Gradient Accumulation for large batch size, batch size scheduling

### ISSUE THREAD

**Issue by netanel-s** [tensorflow/models#5552]
### System information
- **What is the top-level directory of the model you are using**: Object Detection
- **Have I written custom code**: no
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Ubuntu 16.04
- **TensorFlow installed from (source or binary)**: binary
- **TensorFlow version (use command below)**: 1.10.
- **Bazel version (if compiling from source)**: -
- **CUDA/cuDNN version**: CUDA 9
- **GPU model and memory**: 1080Ti, 12GB
- **Exact command to reproduce**:  Not televant

### Describe the problem
I have a request for two features related to batch size. There are a few works which show large batch size is very effective in some computer vision tasks such as object classification and detection (e.g. [1](https://arxiv.org/pdf/1706.02677.pdf), [2](https://arxiv.org/pdf/1711.07240.pdf)). In cases where you don't have multiple-GPU distribution capability (because of lack of hardware or lack of sync capability between the GPUs), this is impossible.

1. Larger batch size that a single GPU can handle can be achieved by Gradient Accumulation, as explained [here](https://medium.com/huggingface/training-larger-batches-practical-tips-on-1-gpu-multi-gpu-distributed-setups-ec88c3e51255) (Large batches on one or several GPU(s)).  This can also be useful in case of multiple-GPUs distribution, where it can simulate further distribution (e.g. 8 GPUs * 8 accumulation steps -> like 64 GPUs in means of batch size).

2. There's also a [paper by Google ](https://arxiv.org/pdf/1711.00489.pdf) which recommends increasing the batch size instead of decreasing the learning rate. 

It would be highly appreciated if you could add these two features.
Thank you very much in advance.

**Comment by ppwwyyxx** [comment]
[My implementation](https://github.com/tensorpack/tensorpack/tree/master/examples/FasterRCNN) of Faster R-CNN and Mask R-CNN uses gradient accumulation when the batch size is small.

**Comment by netanel-s** [comment]
@ppwwyyxx thanks for sharing,
could you please point out the relevant code block in your implementation?
Thanks!

**Comment by ppwwyyxx** [comment]
https://github.com/tensorpack/tensorpack/blob/c842bf51f01b48ffdfefdf47f5ea05912a39dd9a/examples/FasterRCNN/train.py#L72
It's just an optimizer.

**Comment by netanel-s** [comment]
Thanks a lot.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

**Comment by netanel-s** [comment]
> Hi There,
> We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
> If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.

@tensorflowbutler @robieta This issue is still open and relevant to add this feature to the OD API. Thanks!
